File: src/main/java/kr/co/legalai/casework/domain/CaseRecord.java
package kr.co.legalai.casework.domain;

import java.time.Instant;
import java.util.UUID;

public record CaseRecord(
        UUID id,
        String title,
        String status,
        String userPartyRole,
        String userGoal,
        String originalStatement,
        int version,
        Instant createdAt,
        Instant updatedAt
) {
}

File: src/main/kotlin/kr/co/legalai/casework/domain/CaseRecord.kt
package kr.co.legalai.casework.domain

import java.time.Instant
import java.util.UUID

data class CaseRecord(
    val id: UUID,
    val title: String,
    val status: String,
    val userPartyRole: String?,
    val userGoal: String?,
    val originalStatement: String?,
    val version: Int,
    val createdAt: Instant,
    val updatedAt: Instant,
)
